packages/components/src/lib/hooks/useInitialWidth.ts

```diff
@@ -0,0 +1,14 @@
+import { useEffect, useRef, useState } from "react";
+
+export const useInitialWidth = () => {
+  const ref = useRef<HTMLDivElement>(null);
+  const [width, setWidth] = useState(0);
+
+  useEffect(() => {
+    if (ref.current) {
+      setWidth(ref.current.offsetWidth);
+    }
+  }, []); // Empty array means this effect runs once when the component mounts
```

Comment:
Was there a problem with the version I proposed? It used ref-as-a-function, which didn't require useEffect; that's slightly simpler and might be a bit better for performance.

Reply:
I couldn't get it to work. GPT-4 suggested this, it worked. Feel free to try changing. 
(I don't understand hooks very well, so it's very possible I just made some mistake or other)